### DISCUSSION TITLE
Does async `AsyncValue.guard` check for mounted state?

### DISCUSSION THREAD

**Discussion by ggirotto** [rrousselGit/riverpod#3228] [Q&A]
Is it necessary to check for mounted state when using async `AsyncValue.guard` in a `AutoDisposeAsyncNotifier`? And if not, how to protect agains unmount during an async operation in cases like:

```dart
Future<void> asyncFunction() async {
    state = const AsyncValue.loading();
    state = await AsyncValue.guard(() async {
      await asyncOperation();
      // What to do here if the provider is disposed?
    });
  }
```

**Comment by AhmedLSayed9** [comment]
It's necessary but low priority to occur I guess.
Check https://github.com/rrousselGit/riverpod/issues/1879 for how to protect against unmount.
